File: crates/klyron_telemetry/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron::_telemetry {

TelemetryBuilder::TelemetryBuilder() : config_{} {}

TelemetryBuilder& TelemetryBuilder::set_config(const TelemetryConfig& cfg) {
    config_ = cfg;
    return *this;
}

TelemetryConfig TelemetryBuilder::config() const {
    return config_;
}

}

File: crates/klyron_telemetry/bindings/rust/builder.rs
//! Builder for klyron_telemetry
use crate::types::TelemetryConfig;

pub struct TelemetryBuilder {
    config: TelemetryConfig,
}

impl TelemetryBuilder {
    pub fn new() -> Self {
        Self { config: TelemetryConfig::default() }
    }
    pub fn with_version(mut self, version: &str) -> Self {
        self.config.version = version.to_string();
        self
    }
    pub fn build(self) -> TelemetryConfig {
        self.config
    }
}
